webapp/tests/karma/ts/services/get-data-records.service.spec.ts

```diff
@@ -0,0 +1,170 @@
+import { TestBed } from '@angular/core/testing';
+import sinon from 'sinon';
+import { expect } from 'chai';
+
+import { GetDataRecordsService } from '@mm-services/get-data-records.service';
+import { DbService } from '@mm-services/db.service';
+import { HydrateContactNamesService } from '@mm-services/hydrate-contact-names.service';
+import { GetSummariesService } from '@mm-services/get-summaries.service';
+
+describe('GetDataRecords service', () => {
+  let service:GetDataRecordsService;
+  let allDocs;
+  let GetSummaries;
+  let HydrateContactNames;
+
+  beforeEach(() => {
+    allDocs = sinon.stub();
+    GetSummaries = sinon.stub();
+    HydrateContactNames = sinon.stub();
+
+    TestBed.configureTestingModule({
+      providers: [
+        { provide: DbService, useValue: { get: () => ({ allDocs }) } },
+        { provide: HydrateContactNamesService, useValue: { get: HydrateContactNames } },
+        { provide: GetSummariesService, useValue: { get: GetSummaries } },
+      ]
+    });
+
+    service = TestBed.inject(GetDataRecordsService);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+
+  it('returns empty array when given no ids', () => {
```

Comment:
These tests existed in the old code. I didn't want to micro-manage every test with this migration (I think I migrated ~300 unit tests).

Reply:
Actually, I think this is still necessary. Consider this code: 
```
const fn = (item = []) => {
  console.log(item.length);
};

fn(null);
```

We need to null-check anyway, because the default param only replaces undefined.